packages/vsx-registry/src/common/vsx-registry-api.ts

```diff
@@ -107,6 +129,56 @@ export class VSXRegistryAPI {
         return fetchText(url);
     }
 
+    /**
+     * Get the latest compatible extension version.
+     * - an extension satisfies compatibility if its `engines.vscode` version is supported.
+     * @param id the extension id.
+     *
+     * @returns the data for the latest compatible extension version if available, else `undefined`.
+     */
+    async getLatestCompatibleExtensionVersion(id: string): Promise<VSXExtensionRaw | undefined> {
+        const extensions = await this.getAllVersions(id);
+
+        for (let i = 0; i < extensions.length; i++) {
+            const extension: VSXExtensionRaw = extensions[i];
+            if (extension.tags?.includes('Built-in')) {
+                return extension;
```

Comment:
Can we assume that the latest version of a built-in extension can be used with every Theia version?

Reply:
@spoenemann unfortunately, I added this check since some `built-in` extensions were published with an incorrect `engines.vscode` which meant that the extension is incompatible. For example, for `vscode.json-language-features` we have for all versions of the extension:

```json
"engines": {
  "vscode": "0.10.x"
},
```

<details>

<summary>Full Result</summary>

<br />

```
{
  "namespaceUrl": "https://open-vsx.org/api/vscode",
  "reviewsUrl": "https://open-vsx.org/api/vscode/json-language-features/reviews",
  "files": {
    "download": "https://open-vsx.org/api/vscode/json-language-features/1.52.1/file/vscode.json-language-features-1.52.1.vsix",
    "manifest": "https://open-vsx.org/api/vscode/json-language-features/1.52.1/file/package.json",
    "readme": "https://open-vsx.org/api/vscode/json-language-features/1.52.1/file/README.md",
    "license": "https://open-vsx.org/api/vscode/json-language-features/1.52.1/file/LICENSE-vscode.txt",
    "icon": "https://open-vsx.org/api/vscode/json-language-features/1.52.1/file/json.png"
  },
  "name": "json-language-features",
  "namespace": "vscode",
  "version": "1.52.1",
  "publishedBy": {
    "loginName": "open-vsx",
    "fullName": "Open VSX",
    "avatarUrl": "https://avatars0.githubusercontent.com/u/61870837?v=4",
    "homepage": "https://github.com/open-vsx",
    "provider": "github"
  },
  "verified": false,
  "unrelatedPublisher": true,
  "namespaceAccess": "restricted",
  "allVersions": {
    "latest": "https://open-vsx.org/api/vscode/json-language-features/latest",
    "preview": "https://open-vsx.org/api/vscode/json-language-features/preview",
    "1.53.0-next.3ceb3a100e": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.3ceb3a100e",
    "1.53.0-next.ea1b3f27db": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.ea1b3f27db",
    "1.53.0-next.01089c0a50": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.01089c0a50",
    "1.53.0-next.4167837cb8": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.4167837cb8",
    "1.53.0-next.4d8895c7b7": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.4d8895c7b7",
    "1.53.0-next.98106c48a0": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.98106c48a0",
    "1.53.0-next.7242082983": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.7242082983",
    "1.53.0-next.15561c4139": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.15561c4139",
    "1.53.0-next.c44b7d25d9": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.c44b7d25d9",
    "1.53.0-next.5162c8e856": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.5162c8e856",
    "1.53.0-next.caaf2a07a9": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.caaf2a07a9",
    "1.53.0-next.f98fabae6c": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.f98fabae6c",
    "1.53.0-next.6f11d34f67": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.6f11d34f67",
    "1.53.0-next.ec6087fe18": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.ec6087fe18",
    "1.53.0-next.b71972bbb0": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.b71972bbb0",
    "1.53.0-next.c927a8015b": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.c927a8015b",
    "1.53.0-next.d8a7c31aba": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.d8a7c31aba",
    "1.53.0-next.f91b8f2e7a": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.f91b8f2e7a",
    "1.53.0-next.14b7df48c3": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.14b7df48c3",
    "1.53.0-next.96b426ef1d": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.96b426ef1d",
    "1.53.0-next.5954c8366d": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.5954c8366d",
    "1.53.0-next.79557ebbd2": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.79557ebbd2",
    "1.53.0-next.bdd890380e": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.bdd890380e",
    "1.53.0-next.91c7834e92": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.91c7834e92",
    "1.53.0-next.fc9ff5d569": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.fc9ff5d569",
    "1.53.0-next.86d779284b": "https://open-vsx.org/api/vscode/json-language-features/1.53.0-next.86d779284b",
    "1.52.1": "https://open-vsx.org/api/vscode/json-language-features/1.52.1",
    "1.52.0": "https://open-vsx.org/api/vscode/json-language-features/1.52.0",
    "1.52.0-next.532ba43c78": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.532ba43c78",
    "1.52.0-next.ff66544ada": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.ff66544ada",
    "1.52.0-next.5215161c88": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.5215161c88",
    "1.52.0-next.5e54da666c": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.5e54da666c",
    "1.52.0-next.9e5f14c1bc": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.9e5f14c1bc",
    "1.52.0-next.f47aae014c": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.f47aae014c",
    "1.52.0-next.8ee75c19bb": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.8ee75c19bb",
    "1.52.0-next.1e77bd81d8": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.1e77bd81d8",
    "1.52.0-next.5f569de4cd": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.5f569de4cd",
    "1.52.0-next.3898e2ced3": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.3898e2ced3",
    "1.52.0-next.abf3964921": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.abf3964921",
    "1.52.0-next.986e1248f6": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.986e1248f6",
    "1.52.0-next.fb80c0e44a": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.fb80c0e44a",
    "1.52.0-next.ea4d99921c": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.ea4d99921c",
    "1.52.0-next.a89864b295": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.a89864b295",
    "1.52.0-next.d21a51c233": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.d21a51c233",
    "1.52.0-next.d19c7c52fd": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.d19c7c52fd",
    "1.52.0-next.46ff33b6b6": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.46ff33b6b6",
    "1.52.0-next.5afc5cd160": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.5afc5cd160",
    "1.52.0-next.293af2074e": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.293af2074e",
    "1.52.0-next.67817c33a1": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.67817c33a1",
    "1.52.0-next.a28b99d78b": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.a28b99d78b",
    "1.52.0-next.c9bebe2cc9": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.c9bebe2cc9",
    "1.52.0-next.148b85862a": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.148b85862a",
    "1.52.0-next.a2963771d1": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.a2963771d1",
    "1.52.0-next.03c38e6702": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.03c38e6702",
    "1.52.0-next.7f7307b89a": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.7f7307b89a",
    "1.52.0-next.d54c3c67dc": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.d54c3c67dc",
    "1.52.0-next.265a2f6424": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.265a2f6424",
    "1.52.0-next.ad1dc7f87a": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.ad1dc7f87a",
    "1.52.0-next.5264ab94a3": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.5264ab94a3",
    "1.52.0-next.8abfa9bc03": "https://open-vsx.org/api/vscode/json-language-features/1.52.0-next.8abfa9bc03",
    "1.51.1": "https://open-vsx.org/api/vscode/json-language-features/1.51.1",
    "1.51.0": "https://open-vsx.org/api/vscode/json-language-features/1.51.0",
    "1.51.0-next.04863391e9": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.04863391e9",
    "1.51.0-next.7210c7246a": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.7210c7246a",
    "1.51.0-next.a9bbd0c09c": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.a9bbd0c09c",
    "1.51.0-next.9cb11c402f": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.9cb11c402f",
    "1.51.0-next.b56b94a7e4": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.b56b94a7e4",
    "1.51.0-next.1e0df6a4ac": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.1e0df6a4ac",
    "1.51.0-next.cfe2fd1afc": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.cfe2fd1afc",
    "1.51.0-next.ce9ed29d95": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.ce9ed29d95",
    "1.51.0-next.9f7a586124": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.9f7a586124",
    "1.51.0-next.6c8c6008ef": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.6c8c6008ef",
    "1.51.0-next.ff4f319e29": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.ff4f319e29",
    "1.51.0-next.74ef0a92fe": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.74ef0a92fe",
    "1.51.0-next.c13678f258": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.c13678f258",
    "1.51.0-next.89c002ab02": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.89c002ab02",
    "1.51.0-next.a522f37e37": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.a522f37e37",
    "1.51.0-next.72e27c0b23": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.72e27c0b23",
    "1.51.0-next.c0d465ce1c": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.c0d465ce1c",
    "1.51.0-next.78db6d0154": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.78db6d0154",
    "1.51.0-next.276e83dc4b": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.276e83dc4b",
    "1.51.0-next.f5d3ba4d61": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.f5d3ba4d61",
    "1.51.0-next.7a0705650f": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.7a0705650f",
    "1.51.0-next.1f7ba28bde": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.1f7ba28bde",
    "1.51.0-next.c7c471bdde": "https://open-vsx.org/api/vscode/json-language-features/1.51.0-next.c7c471bdde",
    "1.50.1": "https://open-vsx.org/api/vscode/json-language-features/1.50.1",
    "1.50.0": "https://open-vsx.org/api/vscode/json-language-features/1.50.0",
    "1.50.0-next.ed426789c5": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.ed426789c5",
    "1.50.0-next.70a31b6a02": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.70a31b6a02",
    "1.50.0-next.5052a7bc2e": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.5052a7bc2e",
    "1.50.0-next.0fc9b25746": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.0fc9b25746",
    "1.50.0-next.706857dfa5": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.706857dfa5",
    "1.50.0-next.b2e7c41d64": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.b2e7c41d64",
    "1.50.0-next.772aa0723f": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.772aa0723f",
    "1.50.0-next.a9f87e7ca7": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.a9f87e7ca7",
    "1.50.0-next.5d73637a00": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.5d73637a00",
    "1.50.0-next.ddc98c3f62": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.ddc98c3f62",
    "1.50.0-next.a1932df627": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.a1932df627",
    "1.50.0-next.f943318042": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.f943318042",
    "1.50.0-next.fc6b9d1624": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.fc6b9d1624",
    "1.50.0-next.640385a875": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.640385a875",
    "1.50.0-next.868ad44d0f": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.868ad44d0f",
    "1.50.0-next.e8572c0221": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.e8572c0221",
    "1.50.0-next.1343efabed": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.1343efabed",
    "1.50.0-next.b91c581dd9": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.b91c581dd9",
    "1.50.0-next.e13875b77c": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.e13875b77c",
    "1.50.0-next.cca20eba89": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.cca20eba89",
    "1.50.0-next.ee11170fef": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.ee11170fef",
    "1.50.0-next.03960a5aa1": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.03960a5aa1",
    "1.50.0-next.940bfbd6a6": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.940bfbd6a6",
    "1.50.0-next.1d7b32b880": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.1d7b32b880",
    "1.50.0-next.0d8cc5e56d": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.0d8cc5e56d",
    "1.50.0-next.6c797984fb": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.6c797984fb",
    "1.50.0-next.892353d48e": "https://open-vsx.org/api/vscode/json-language-features/1.50.0-next.892353d48e",
    "1.49.3": "https://open-vsx.org/api/vscode/json-language-features/1.49.3",
    "1.49.2": "https://open-vsx.org/api/vscode/json-language-features/1.49.2",
    "1.49.1": "https://open-vsx.org/api/vscode/json-language-features/1.49.1",
    "1.49.0": "https://open-vsx.org/api/vscode/json-language-features/1.49.0",
    "1.49.0-next.45c70c29e5": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.45c70c29e5",
    "1.49.0-next.7eaf220caf": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.7eaf220caf",
    "1.49.0-next.c7c6c5cde1": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.c7c6c5cde1",
    "1.49.0-next.98fe9425f2": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.98fe9425f2",
    "1.49.0-next.61d8097124": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.61d8097124",
    "1.49.0-next.b96ce60e2d": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.b96ce60e2d",
    "1.49.0-next.c6e64d2217": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.c6e64d2217",
    "1.49.0-next.1d7ff4c718": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.1d7ff4c718",
    "1.49.0-next.b58f4549b2": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.b58f4549b2",
    "1.49.0-next.0f22246a44": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.0f22246a44",
    "1.49.0-next.3ab605c4de": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.3ab605c4de",
    "1.49.0-next.f57a418b01": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.f57a418b01",
    "1.49.0-next.ad40702857": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.ad40702857",
    "1.49.0-next.c50b5acd37": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.c50b5acd37",
    "1.49.0-next.c85c46579e": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.c85c46579e",
    "1.49.0-next.64d126cafb": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.64d126cafb",
    "1.49.0-next.f694b07825": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.f694b07825",
    "1.49.0-next.650f7688f3": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.650f7688f3",
    "1.49.0-next.40a54b0e33": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.40a54b0e33",
    "1.49.0-next.0a22f70e66": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.0a22f70e66",
    "1.49.0-next.e0cc8f579f": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.e0cc8f579f",
    "1.49.0-next.95b3a0a8a0": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.95b3a0a8a0",
    "1.49.0-next.f7f05dee53": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.f7f05dee53",
    "1.49.0-next.c17b4c2488": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.c17b4c2488",
    "1.49.0-next.9bd54c15ae": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.9bd54c15ae",
    "1.49.0-next.466da1c901": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.466da1c901",
    "1.49.0-next.6d913c4c14": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.6d913c4c14",
    "1.49.0-next.11c0b60785": "https://open-vsx.org/api/vscode/json-language-features/1.49.0-next.11c0b60785",
    "1.48.2": "https://open-vsx.org/api/vscode/json-language-features/1.48.2",
    "1.48.1": "https://open-vsx.org/api/vscode/json-language-features/1.48.1",
    "1.48.0": "https://open-vsx.org/api/vscode/json-language-features/1.48.0",
    "1.48.0-next.42a9a9998e": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.42a9a9998e",
    "1.48.0-next.f0283ed239": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.f0283ed239",
    "1.48.0-next.d762190a09": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.d762190a09",
    "1.48.0-next.d933c49a5c": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.d933c49a5c",
    "1.48.0-next.d5327bca82": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.d5327bca82",
    "1.48.0-next.eb05e94d5e": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.eb05e94d5e",
    "1.48.0-next.8d51d4a419": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.8d51d4a419",
    "1.48.0-next.cf1b357443": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.cf1b357443",
    "1.48.0-next.c2224dab6c": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.c2224dab6c",
    "1.48.0-next.3bb6bc5bcd": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.3bb6bc5bcd",
    "1.48.0-next.9d72fcb0ae": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.9d72fcb0ae",
    "1.48.0-next.8ca1cf6293": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.8ca1cf6293",
    "1.48.0-next.e2de23c683": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.e2de23c683",
    "1.48.0-next.c53cc6546f": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.c53cc6546f",
    "1.48.0-next.db170566b8": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.db170566b8",
    "1.48.0-next.bae379e986": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.bae379e986",
    "1.48.0-next.4a1bcdafe9": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.4a1bcdafe9",
    "1.48.0-next.0bf4493b75": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.0bf4493b75",
    "1.48.0-next.d586f701ec": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.d586f701ec",
    "1.48.0-next.6c21258b65": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.6c21258b65",
    "1.48.0-next.2c609ec4aa": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.2c609ec4aa",
    "1.48.0-next.7099dc0708": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.7099dc0708",
    "1.48.0-next.c24de77fe8": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.c24de77fe8",
    "1.48.0-next.cbbb8c7665": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.cbbb8c7665",
    "1.48.0-next.b8e773ebed": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.b8e773ebed",
    "1.48.0-next.aa930e5619": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.aa930e5619",
    "1.48.0-next.3d4ff091be": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.3d4ff091be",
    "1.48.0-next.1ca18ed83c": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.1ca18ed83c",
    "1.48.0-next.ae34596ac2": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.ae34596ac2",
    "1.48.0-next.5eb91c7c40": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.5eb91c7c40",
    "1.48.0-next.46c2494cb0": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.46c2494cb0",
    "1.48.0-next.729343feab": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.729343feab",
    "1.48.0-next.4e4941b802": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.4e4941b802",
    "1.48.0-next.ad8c805aa6": "https://open-vsx.org/api/vscode/json-language-features/1.48.0-next.ad8c805aa6",
    "1.47.3": "https://open-vsx.org/api/vscode/json-language-features/1.47.3",
    "1.47.2": "https://open-vsx.org/api/vscode/json-language-features/1.47.2",
    "1.47.1": "https://open-vsx.org/api/vscode/json-language-features/1.47.1",
    "1.47.0": "https://open-vsx.org/api/vscode/json-language-features/1.47.0",
    "1.47.0-next.e38c2d293d": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.e38c2d293d",
    "1.47.0-next.5dc7025c1e": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.5dc7025c1e",
    "1.47.0-next.ef903c70c0": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.ef903c70c0",
    "1.47.0-next.b26cde750e": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.b26cde750e",
    "1.47.0-next.7923112cdd": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.7923112cdd",
    "1.47.0-next.04c36be045": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.04c36be045",
    "1.47.0-next.3158fc0f76": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.3158fc0f76",
    "1.47.0-next.c7ea388c7d": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.c7ea388c7d",
    "1.47.0-next.c17ed9ffec": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.c17ed9ffec",
    "1.47.0-next.61f796ee89": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.61f796ee89",
    "1.47.0-next.fad4d8d254": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.fad4d8d254",
    "1.47.0-next.9c142ea383": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.9c142ea383",
    "1.47.0-next.97650fb8f3": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.97650fb8f3",
    "1.47.0-next.8cd3fe080b": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.8cd3fe080b",
    "1.47.0-next.6c68514534": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.6c68514534",
    "1.47.0-next.ca7f2683fd": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.ca7f2683fd",
    "1.47.0-next.9dde49dc78": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.9dde49dc78",
    "1.47.0-next.af9d0d5df2": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.af9d0d5df2",
    "1.47.0-next.1a1ab8a435": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.1a1ab8a435",
    "1.47.0-next.a6e747b734": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.a6e747b734",
    "1.47.0-next.0145e38e5f": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.0145e38e5f",
    "1.47.0-next.2aa9854ac4": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.2aa9854ac4",
    "1.47.0-next.102311f877": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.102311f877",
    "1.47.0-next.4d6da3154c": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.4d6da3154c",
    "1.47.0-next.5347e157ef": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.5347e157ef",
    "1.47.0-next.6fef59565b": "https://open-vsx.org/api/vscode/json-language-features/1.47.0-next.6fef59565b",
    "1.46.1": "https://open-vsx.org/api/vscode/json-language-features/1.46.1",
    "1.46.0": "https://open-vsx.org/api/vscode/json-language-features/1.46.0",
    "1.46.0-next.cde20069b3": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.cde20069b3",
    "1.46.0-next.11848b40d3": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.11848b40d3",
    "1.46.0-next.d760fde11e": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.d760fde11e",
    "1.46.0-next.6532cc6c8d": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.6532cc6c8d",
    "1.46.0-next.ce14facbe7": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.ce14facbe7",
    "1.46.0-next.79a1f5a5ca": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.79a1f5a5ca",
    "1.46.0-next.50f907f0ba": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.50f907f0ba",
    "1.46.0-next.e70bf91f55": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.e70bf91f55",
    "1.46.0-next.028241b160": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.028241b160",
    "1.46.0-next.40559425c7": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.40559425c7",
    "1.46.0-next.040034f7ae": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.040034f7ae",
    "1.46.0-next.21d6116577": "https://open-vsx.org/api/vscode/json-language-features/1.46.0-next.21d6116577",
    "1.45.1": "https://open-vsx.org/api/vscode/json-language-features/1.45.1",
    "1.44.2": "https://open-vsx.org/api/vscode/json-language-features/1.44.2",
    "1.39.1": "https://open-vsx.org/api/vscode/json-language-features/1.39.1"
  },
  "downloadCount": 72773,
  "reviewCount": 0,
  "versionAlias": [
    "latest"
  ],
  "timestamp": "2020-12-17T00:22:56.691122Z",
  "preview": false,
  "displayName": "JSON Language Features (built-in)",
  "description": "Provides rich language support for JSON files.",
  "engines": {
    "vscode": "0.10.x"
  },
  "categories": [
    "Programming Languages"
  ],
  "tags": [
    "Built-in"
  ],
  "license": "MIT",
  "repository": "https://github.com/theia-ide/vscode-builtin-extensions",
  "dependencies": [],
  "bundledExtensions": []
}
```
</details>

Because of this incorrect `engines.vscode` we perform multiple requests on startup for all versions of the extension (which slows down the application). Since the builtins are controlled by application developers I thought this might be a way around the issue. I'm not sure if there is something we can do on `open-vsx` to resolve the problem and re-publish.

Reply:
Republishing might be an option. @marcdumais WDYT?

Reply:
> Republishing might be an option

It might be but I think some help will be required from the registry's side, since normally `ovsx` will refuse to re-publish already existing extensions. Then there's a bit of a risk that extensions that built and worked a while ago may no longer. 

Alternatively we could add intelligence in the registry or the client, I think. The original build-ins, as they are included in vscode, do not require to publicize their `engines.vscode` since it should always match the version of vscode they come along-with (same repo). The idea would be for the server or client to have special handling, taking for granted that builtin version == `engines.vscode`. 

WDYT?

Reply:
We could generalize this:
```typescript
if (!extension.engines) {
    return extension;
}
```

Reply:
@marcdumais-work @spoenemann I just wanted to mention that for certain builtins (such as `vscode.json-language-features`) the api version is incorrect from vscode (specified as `0.10.x`).

- https://github.com/microsoft/vscode/blob/37c162ed6df0096d658d89284847bf075acc66dd/extensions/json-language-features/package.json#L9-L11

The logic to skip the compatibility check for `builtins` that this pull-request includes is something similar to the behavior vscode has as well to skip builtins:

- https://github.com/microsoft/vscode/blob/37c162ed6df0096d658d89284847bf075acc66dd/src/vs/platform/extensions/common/extensionValidator.ts#L214-L222

Reply:
>  for certain builtins (such as vscode.json-language-features) the api version is incorrect

Indeed. I think some extensions that are currently builtins were previously standalone, and may have retained that field, and not maintained it. Others do not have the field at all as per the above comment. Either way the field is not needed for the primary intended use of these extensions. 

Could we instead distinguish these extensions by publisher?

Reply:
@marcdumais-work would you rather me exclude `vscode` as a publisher, at the moment I'm using the `Built-in` keyword which is used when we publish from the `vscode-builtin-extensions`:

- https://github.com/theia-ide/vscode-builtin-extensions/blob/0e1f562e0e3fb8b514bd03bb0ae46f1793c5a5b5/src/package-vsix.js#L131

Reply:
The publisher (namespace) name is a safer criterion because it is protected. Anyone could publish an extension and add the `Built-in` keyword.